### PR TITLE
cask_language_reference: specify new rules for name

### PR DESCRIPTION
**Pull request by vitorgalvao**

### CODE REVIEW

File: doc/CASK_LANGUAGE_REFERENCE.md [hunk 1]
```diff
@@ -178,11 +178,11 @@ There are currently some arbitrary limitations on Cask tokens which are in the p
 
 ## Name Stanza Details
 
-`name` stanza accepts a UTF-8 string defining the full name of the software.
+`name` accepts a UTF-8 string defining the full name of the software, and is used to help with searchability and disambiguation. It can be repeated multiple times if there are useful alternative names.
 
-If there are useful alternate names, `name` can be repeated multiple times. (Or, equivalently, an array value may be given.)
+Its first instance should use the latin alphabet and include the software vendor’s name and be the as verbose as possible while still making sense.
```
Comment (Amorymeltzer) [inline doc/CASK_LANGUAGE_REFERENCE.md]: Grammar-wise, should read:

`Its first instance should use the latin alphabet, include the software vendor’s name, and be as verbose as possible while still making sense.`

File: doc/CASK_LANGUAGE_REFERENCE.md [hunk 2]
```diff
@@ -178,11 +178,11 @@ There are currently some arbitrary limitations on Cask tokens which are in the p
...
 
-When multiple names are given, the first should follow the canonical branding as defined by the vendor.
+A good example is [pycharm](link to specific revision). `Jetbrains PyCharm Community Edition` makes sense even though it is likely never referenced as such anywhere, but `Jetbrains PyCharm Community Edition CE` doesn’t, hence why it has a second line. Another example are casks whose original name does not use the latin alphabet, like [cave-story](link to specific revision).
```
Comment (Amorymeltzer) [inline doc/CASK_LANGUAGE_REFERENCE.md]: Should be `pycharm-ce`; `pycharm` is a different cask.

Comment (vitorgalvao) [inline doc/CASK_LANGUAGE_REFERENCE.md]: Updated.

**Comment by adityadalal924** [comment]
:+1 after these changes

**Pull request merged**